src/cljs/main/broadfcui/common/components.cljs

```diff
@@ -186,61 +186,72 @@
-              [CodeMirror {:text (get-in entity [:method :payload])}]]
-             [CodeMirror {:text (:payload entity)}]))])])
+         (when-not redacted?
+           [:div {:style {:paddingTop "0.5rem"}}
+            [:span {:style {:fontWeight 500 :marginRight "1rem"}} (if config? "Referenced Method:" "WDL:")]
+            (links/create-internal {:onClick #(swap! state update :payload-expanded not)}
+                                   (if (:payload-expanded @state) "Collapse" "Expand"))]
+           (when (:payload-expanded @state)
+             (if config?
+               [:div {:style {:margin "0.5rem 0 0 1rem"}}
+                (this :render-details (:method entity))
+                [:div {:style {:fontWeight 500 :marginTop "1rem"}} "WDL:"]
+                [CodeMirror {:text (get-in entity [:method :payload])}]]
+               [CodeMirror {:text (:payload entity)}])))])])
    :render-details
-   (fn [{:keys [props refs]} entity]
-     (let [{:keys [editing?]} props
+   (fn [{:keys [props refs state]} entity]
+     (let [{:keys [editing? redacted?]} props
+           original-redacted? (:original-redacted? @state)
            make-field
            (fn [key label & {:keys [dropdown? wrap? render]}]
              [:div {:style {:display "flex" :alignItems "baseline" :paddingBottom "0.25rem"}}
               [:div {:style {:flex "0 0 100px" :fontWeight 500}} (str label ":")]
               [:div {:style {:flex "1 1 auto" :overflow "hidden" :textOverflow "ellipsis"
                              :whiteSpace (when-not wrap? "nowrap")}}
                (if (and editing? dropdown?)
-                 (style/create-identity-select {:ref key
-                                                :data-test-id (config/when-debug "edit-method-config-snapshot-id-select")
-                                                :style {:width 100}
-                                                :defaultValue (key entity)
-                                                :onChange (when-let [f (:onSnapshotIdChange props)]
-                                                            #(f (int (common/get-text refs "snapshotId"))))}
-                                               (:snapshots props))
+                 (style/create-identity-select-name {:ref key
+                                                     :data-test-id (config/when-debug "edit-method-config-snapshot-id-select")
+                                                     :style {:width 120}
+                                                     :defaultValue (if original-redacted? -1 (key entity))
+                                                     :onChange (when-let [f (:onSnapshotIdChange props)]
+                                                                 #(f (int (common/get-text refs "snapshotId"))))}
+                                                    (:snapshots props)
+                                                    (when original-redacted? original-redacted?))
```

Comment:
ah, it initially was a boolean, but then it changed to be the redacted version, and then the display string. i'll remove the ?